core/src/main/java/tc/oc/pgm/community/events/PlayerVanishEvent.java

```diff
@@ -0,0 +1,35 @@
+package tc.oc.pgm.community.events;
+
+import org.bukkit.event.HandlerList;
+import tc.oc.pgm.api.event.ExtendedCancellable;
+import tc.oc.pgm.api.player.MatchPlayer;
+
+public class PlayerVanishEvent extends ExtendedCancellable {
```

Comment:
Make sure to extend existing events, seems like this would fall under `MatchPlayerEvent`